valid-form.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+$logfile = fopen("logfile.txt", "w");
+
+$valid = array();
+$errors = array();
+
+if ($_POST) {
+$title = $_POST['title'];
+$lecturer = $_POST['lecturer'];
+$description = $_POST['description'];
+$group = $_POST['group'];
+$credits = $_POST['credits'];
+
+if (!$title) {
+$errors['title'] = 'Име на предмета е задължително поле.';
+} elseif (strlen($title) > 150) {
+$errors['title'] = 'Името на предмета има максимална дължина 150 символа.';
+} else {
+$valid['title'] = $title;
+}
+
+if (!$lecturer) {
+$errors['lecturer'] = 'Преподавател е задължително поле.';
+} elseif (strlen($lecturer) > 200) {
+$errors['lecturer'] = 'Преподавател има максимална дължина 200 символа.';
+} else {
+$valid['lecturer'] = $lecturer;
+}
+
+if (!$description) {
+$errors['description'] = 'Описание е задължително поле.';
+} elseif (strlen($description) < 10) {
```

Comment:
Please extract all constant values into constants/variables and give them meaningful names. This way you will know what is the purpose of the value when reading the code. For example here you would be able to reuse the value of 10 in the error message too here that way.